vms/platformvm/txs/executor/proposal_tx_executor.go

```diff
@@ -400,6 +402,72 @@ func (e *ProposalTxExecutor) RewardValidatorTx(tx *txs.RewardValidatorTx) error
 
 			e.OnCommitState.AddUTXO(utxo)
 			e.OnCommitState.AddRewardUTXO(tx.TxID, utxo)
+
+			offset++
+		}
+
+		// Provide the accrued delegatee rewards from successful delegations here.
+		delegateeReward, err := e.OnCommitState.GetDelegateeReward(
+			stakerToRemove.SubnetID,
+			stakerToRemove.NodeID,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to fetch accrued delegatee rewards: %w", err)
+		}
+
+		if delegateeReward > 0 {
+			delegationRewardsOwner := uStakerTx.DelegationRewardsOwner()
+			outIntf, err := e.Fx.CreateOutput(delegateeReward, delegationRewardsOwner)
+			if err != nil {
+				return fmt.Errorf("failed to create output: %w", err)
+			}
+			out, ok := outIntf.(verify.State)
+			if !ok {
+				return errInvalidState
+			}
+			utxo := &avax.UTXO{
+				UTXOID: avax.UTXOID{
+					TxID:        tx.TxID,
+					OutputIndex: uint32(len(outputs) + len(stake) + offset),
+				},
+				Asset: stakeAsset,
+				Out:   out,
+			}
+
+			e.OnCommitState.AddUTXO(utxo)
+			e.OnCommitState.AddRewardUTXO(tx.TxID, utxo)
+		}
+
+		// there is no [offset] for the accrued delegatee rewards if the vdr tx is aborted
+		delegateeReward, err = e.OnAbortState.GetDelegateeReward(
```

Comment:
I don't think we should need to be reading the `OnAbortState` here right? I think we should be able to reuse the `out` for the onCommitUTXO and the onAbortUTXO.